docs/docs/guides/web3_plugin_guide/plugin_authors.md

```diff
@@ -0,0 +1,212 @@
+---
+sidebar_position: 1
+sidebar_label: 'Plugin Authors'
+---
+
+# Web3.js Plugin Author's Guide
+
+This guide intends to provide the necessary context for developing plugins for Web3.js.
+
+## Before Getting Started
+
+It's highly recommended you as the plugin author understand the limitations of TypeScript's module augmentation as described in the [main plugin guide](/docs/guides/web3_plugin_guide/), so you can communicate to your users that they are responsible for augmenting the class interface they register your plugin with if they desire to have type support when using your plugin. Ideally this could be solved for by the plugin author, or better yet Web3.js, but so far a better solution is unknown - if you have any ideas, please [create an issue](https://github.com/web3/web3.js/issues/new/choose) and help us improve Web3.js' UX.
```

Comment:
```suggestion
It's highly recommended you as the plugin author understand the limitations of TypeScript's module augmentation as described in the [main plugin guide](/docs/guides/web3_plugin_guide/), so you can communicate to your users that they are responsible for augmenting the class interface they register your plugin with if they desire to have type support when using your plugin. 
```

For this part we should create a github discussion instead of placing in this doc.

Reply:
I think it should be in the doc as it where the issue is most relevant, but we could create a discussion and link it here